diff --git a/node_modules/three-stdlib/controls/OrbitControls.js b/node_modules/three-stdlib/controls/OrbitControls.js
index 23e5b19..5b49099 100644
--- a/node_modules/three-stdlib/controls/OrbitControls.js
+++ b/node_modules/three-stdlib/controls/OrbitControls.js
@@ -98,6 +98,7 @@ class OrbitControls extends EventDispatcher {
     this.target0 = this.target.clone();
     this.position0 = this.object.position.clone();
     this.zoom0 = this.object.zoom;
+    this.minPanDistance = 0;
     this.getPolarAngle = () => spherical.phi;
     this.getAzimuthalAngle = () => spherical.theta;
     this.setPolarAngle = (value) => {
@@ -366,6 +367,9 @@ class OrbitControls extends EventDispatcher {
       return function panLeft2(distance, objectMatrix) {
         v.setFromMatrixColumn(objectMatrix, 0);
         v.multiplyScalar(-distance);
+        if (Math.abs(distance) < scope.minPanDistance) {
+          return;
+        }
         panOffset.add(v);
       };
     })();
@@ -379,6 +383,9 @@ class OrbitControls extends EventDispatcher {
           v.crossVectors(scope.object.up, v);
         }
         v.multiplyScalar(distance);
+        if (Math.abs(distance) < scope.minPanDistance) {
+          return;
+        }
         panOffset.add(v);
       };
     })();
