diff --git a/node_modules/three-stdlib/controls/OrbitControls.js b/node_modules/three-stdlib/controls/OrbitControls.js
index 23e5b19..36ba747 100644
--- a/node_modules/three-stdlib/controls/OrbitControls.js
+++ b/node_modules/three-stdlib/controls/OrbitControls.js
@@ -366,6 +366,9 @@ class OrbitControls extends EventDispatcher {
       return function panLeft2(distance, objectMatrix) {
         v.setFromMatrixColumn(objectMatrix, 0);
         v.multiplyScalar(-distance);
+        if (v.lengthSq() < 0.03 ** 2) {
+          return;
+        }
         panOffset.add(v);
       };
     })();
@@ -379,6 +382,9 @@ class OrbitControls extends EventDispatcher {
           v.crossVectors(scope.object.up, v);
         }
         v.multiplyScalar(distance);
+        if (v.lengthSq() < 0.03 ** 2) {
+          return;
+        }
         panOffset.add(v);
       };
     })();
